### DISCUSSION TITLE
How to calculate AutoV1?

### DISCUSSION THREAD

**Discussion by Dingo64** [civitai/civitai#153] [General]
What is the algorithm?

**Comment by Dingo64** [comment]
OK, I found out by trial & error and some digging.

AutoV1 is Automatic1111's hashing algo version 1, which in PHP can be coded like this:

```
$chunk = file_get_contents($fname, false, null, 1048576, 65536);
$hash = hash('sha256', $chunk);
$hash = substr($hash, 0, 8);
```

We are getting 65536 bytes starting at offset 1048576 and performing SHA-256 on it. Then from the resulting hash we copy first 4 bytes and display as 8 hex digits.

Now- the hash on Civitai is shown for the SafeTensors version od the model and the PickleTensors version has a different hash (which is not shown on Civitai).

**Comment by JustMaier** [comment]
Since you posted, Auto has moved to a new hash standard, we're calling it Auto V2. Its just the first 10 characters of the SHA256 hash of a file. I would expect this to mean that the SafeTensors version will have a different hash than the same model as a ckpt.
